src/MSXMotherBoard.cc

```diff
@@ -389,50 +390,68 @@ string MSXMotherBoard::loadMachine(const string& machine)
 	return machineName;
 }
 
-void MSXMotherBoard::storeAsSetup(const string& filename)
+void MSXMotherBoard::storeAsSetup(const string& filename, SetupDepth depth)
 {
+	// level 0: don't do anything. Added as convenience.
+	if (depth == SetupDepth::NONE) return;
+
+	XmlOutputArchive out(filename);
+
+	if (depth == SetupDepth::COMPLETE_STATE) {
+		// level 5: just save state to given file
+		out.serialize("machine", this);
+		out.close();
+		return;
+	}
+
 	// level 1: create new board based on current board of this machine
 	auto newBoard = reactor.createEmptyMotherBoard();
 	newBoard->loadMachine(machineName);
```

Comment:
Not introduced in this commit, but I only notice it now.
For a moment I thought this could crash, but now I think it's fine. No need to do anything, but I'll explain anyway:

In theory this step can fail (throw an exception). When the machine config file was removed between the time the current machine was started and when the save_setup command was executed. I'd guess this will throw a FileNotFoundException, and my fear was that `Interpreter::commandProc` would not catch this (I thought it would only catch `CommandException`), but luckily that's not the case, it will catch all `MSXException` types.

Reply:
Ah, well spotted. I probably didn't realize this either, but thanks for making it clear. And glad it's OK like this.